### PR TITLE
NOX-814 Update Put Owned Entity (Create and Update) Localized API

### PR DESCRIPTION
**Pull request by msadikkoseaupa**

### CODE REVIEW

File: src/Nox.Generator/Presentation/Api/OData/EntityController.Translations.template.cs [hunk 1]
```diff
@@ -107,14 +107,18 @@ public abstract partial class {{className}}Base
...
+    public virtual async Task<ActionResult<{{GetEntityDtoNameForLocalizedType localizedRelationship.OwnedEntity.Name}}>> Put{{localizedRelationship.OwnedEntity.Name}}Localized({{ primaryKeysRoute }}, {{- if localizedRelationship.IsWithMultiEntity }}{{localizedRelationship.OwnedEntityKeysRoute}},{{end}} [FromRoute] System.String {{ cultureCode}}, [FromBody] {{ GetEntityUpsertDtoNameForLocalizedType localizedRelationship.OwnedEntity.Name}} {{ToLowerFirstChar localizedRelationship.OwnedEntity.Name}}LocalizedUpsertDto)
     {
         if (!ModelState.IsValid)
         {
             throw new Nox.Exceptions.BadRequestException(ModelState);
         }
+        Nox.Exceptions.BadRequestException.ThrowIfNotValid(Nox.Types.CultureCode.TryFrom({{cultureCode}}, out var {{cultureCode}}Value));
         
+        {{- if localizedRelationship.IsWithMultiEntity }}
+        {{localizedRelationship.OwnedEntityKey}} = relatedKey;
```
Comment (dejobratic) [inline src/Nox.Generator/Presentation/Api/OData/EntityController.Translations.template.cs:120]: Will this work if the owned entity has compound key? 

Here is how we set the OwnedEnityKey property in the generator:
image

**Review by dejobratic — COMMENTED** [review]
Looks good.

File: tests/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Presentation.Api.OData/TenantsController/Translations.g.cs [hunk 2]
```diff
@@ -76,14 +77,14 @@ public virtual async Task<ActionResult<TenantBrandLocalizedDto>> DeleteTenantBra
 
         return NoContent();
     }
-    [HttpPut("/api/v1/Tenants/{key}/TenantContactLocalized/{cultureCode}")]
-    public virtual async Task<ActionResult<TenantContactLocalizedDto>> PutTenantContactLocalized( [FromRoute] System.UInt32 key, [FromRoute] System.String cultureCode, [FromBody] TenantContactLocalizedUpsertDto tenantContactLocalizedUpsertDto)
+    [HttpPut("/api/v1/Tenants/{key}/TenantContacts/Languages/{cultureCode}")]
```
Comment (rochar) [inline tests/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Presentation.Api.OData/TenantsController/Translations.g.cs]: Singular?
```suggestion
    [HttpPut("/api/v1/Tenants/{key}/TenantContact/Languages/{cultureCode}")]
```

**Review by rochar — CHANGES_REQUESTED** [review]
looks good, a minor fix

File: src/Nox.Generator/Presentation/Api/OData/EntityController.Translations.template.cs (hunk 1 above)
Comment (msadikkoseaupa) [reply src/Nox.Generator/Presentation/Api/OData/EntityController.Translations.template.cs:120]: Indeed, owned entity does not support compound keys, so I updated this part.

File: tests/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Presentation.Api.OData/TenantsController/Translations.g.cs (hunk 2 above)
Comment (msadikkoseaupa) [reply tests/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Presentation.Api.OData/TenantsController/Translations.g.cs]: Desired changes have been applied.

**Review by dejobratic — APPROVED** [review]

**Review by rochar — APPROVED** [review]

**Pull request merged**